Distinguish SCCs that AllowHostNetwork and AllowHostPorts from those that do not, in the score calculation.

adelton opened the pull request:
Fixes https://github.com/openshift/origin/issues/15933.

adelton commented:
/retest

php-coder commented on pkg/cmd/server/bootstrappolicy/securitycontextconstraints_test.go:
```diff
@@ -16,8 +16,8 @@ func TestBootstrappedConstraints(t *testing.T) {
-		SecurityContextConstraintsHostNetwork,
 		SecurityContextConstraintRestricted,
+		SecurityContextConstraintsHostNetwork,
 		SecurityContextConstraintNonRoot,
```
I'd expect that `nonroot` is stricter than `hostnetwork`. Hm. Is it because `nonroot` doesn't require to drop any capabilities?

adelton replied:
`SecurityContextConstraintNonRoot` in https://github.com/openshift/origin/blob/master/pkg/cmd/server/bootstrappolicy/securitycontextconstraints.go#L92 has `securityapi.RunAsUserStrategyMustRunAsNonRoot` while `SecurityContextConstraintsHostNetwork` in https://github.com/openshift/origin/blob/master/pkg/cmd/server/bootstrappolicy/securitycontextconstraints.go#L240 has `securityapi.RunAsUserStrategyMustRunAsRange`.

And in https://github.com/openshift/origin/blob/master/pkg/security/securitycontextconstraints/byrestrictions.go#L72 and https://github.com/openshift/origin/blob/master/pkg/security/securitycontextconstraints/byrestrictions.go#L37, the values are

```
runAsNonRootPoints points = 30000
runAsRangePoints points = 20000
```

That's why the `hostnetwork` with `AsRange` ends up being less strict than `nonroot` with `AsNonRoot`.

Of course, the question is why `runAsRangePoints` < `runAsNonRootPoints` which has been that way since 6db81779c852b5285b6b07ede964ec8436a5dc64.

adelton replied:
Hmm, that was just a renaming commit. The original logic comes from 89d91534576bf23c5d8ca77b830a17e0a86bd180. @pweil-, would you remember why it had

```
+       case kapi.RunAsUserStrategyMustRunAsNonRoot:
+               points += 3
+       case kapi.RunAsUserStrategyMustRunAsRange:
+               points += 2
```

?

pweil- replied:
non-root is more "relaxed" because it is up to the user.  MustRunAsRange means the admin has made a decision on the user's behalf.  That said, the fact that non-root is not dropping caps is something that raises my eyebrow.  

Anything that allows host access should be considered more dangerous that things that don't.  Separate bug though than what @adelton is addressing here.

php-coder replied:
> That said, the fact that non-root is not dropping caps is something that raises my eyebrow.

I've created https://github.com/openshift/origin/issues/16371 for this.

adelton commented:
> Anything that allows host access should be considered more dangerous that things that don't.

Thanks. Moved host network and host port right below privileged in the score calculation in new proposal.

Rebased on master -> 971e472b63b16c53c7279d220466805380d61ebd.

adelton commented:
/retest

adelton commented:
/test extended_conformance_gce
/test cmd

php-coder commented:
@openshift/sig-security PTAL

enj commented:
/approve

enj commented:
/kind bug

enj commented:
@php-coder LGTM if you are happy.

php-coder commented:
/lgtm

@adelton Thank you!

php-coder commented:
@smarterclayton @pweil- This PR changes the order of the SCCs. Perhaps, we should mention it in the release notes as it could affect users after upgrade.

php-coder commented:
/remove-kind bug
